spring-cloud-contract-tools/spring-cloud-contract-spec-pact/src/main/groovy/org/springframework/cloud/contract/verifier/spec/pact/MatchingRulesConverter.groovy

```diff
@@ -0,0 +1,68 @@
+package org.springframework.cloud.contract.verifier.spec.pact
+
+import au.com.dius.pact.model.matchingrules.Category
+import au.com.dius.pact.model.matchingrules.EqualsMatcher
+import au.com.dius.pact.model.matchingrules.MaxTypeMatcher
+import au.com.dius.pact.model.matchingrules.MinMaxTypeMatcher
+import au.com.dius.pact.model.matchingrules.MinTypeMatcher
+import au.com.dius.pact.model.matchingrules.RegexMatcher
+import au.com.dius.pact.model.matchingrules.TypeMatcher
+import groovy.transform.CompileStatic
+import groovy.transform.PackageScope
+import org.springframework.cloud.contract.spec.internal.BodyMatcher
+import org.springframework.cloud.contract.spec.internal.BodyMatchers
+import org.springframework.cloud.contract.spec.internal.MatchingType
+
+/**
+ * @author Tim Ysewyn
+ * @Since 2.0.0
+ */
+@CompileStatic
+@PackageScope
+class MatchingRulesConverter {
+
+	static Category matchingRulesForBody(BodyMatchers bodyMatchers) {
+		matchingRulesFor("body", bodyMatchers)
+	}
+
+	private static Category matchingRulesFor(String categoryName, BodyMatchers bodyMatchers) {
+		Category category = new Category(categoryName)
+
+		bodyMatchers.jsonPathMatchers().forEach({ BodyMatcher it ->
+			String key = getMatcherKey(it.path())
+			MatchingType matchingType = it.matchingType()
+
+			switch (matchingType) {
+				case MatchingType.EQUALITY:
+					category.setRule(key, EqualsMatcher.INSTANCE)
+					break
+				case MatchingType.TYPE:
+					if (it.minTypeOccurrence() && it.maxTypeOccurrence()) {
+						category.setRule(key, new MinMaxTypeMatcher(it.minTypeOccurrence(), it.maxTypeOccurrence()))
+					} else if (it.minTypeOccurrence()) {
+						category.setRule(key, new MinTypeMatcher(it.minTypeOccurrence()))
+					} else if (it.maxTypeOccurrence()) {
+						category.setRule(key, new MaxTypeMatcher(it.maxTypeOccurrence()))
+					} else {
+						category.setRule(key, TypeMatcher.INSTANCE)
+					}
+					break
+				case MatchingType.DATE:
```

Comment:
There are no longer such rules in Pact? I added them specifically cause they were in pact.

Reply:
Oh they are, for some reason I looked over it.
And I will need to add 4 more for v3: INCLUDE, NUMBER, DECIMAL and NULL